### ISSUE TITLE
Improve install app experience when already logged in

### ISSUE THREAD

**Issue by nickmealey** [Shopify/shopify_app#1104]
It seems that since adding user tokens to authentication, we're running into an issue where if a user is already logged into the app and attempts to install on a different shop (via the app store, for example), the install will fail. This is only an issue when both `shop_session_repository` and `user_session_repository` are enabled.

The reason it was failing in our case was that we're using `after_authenticate_job` inline, and it is passed a `shop_domain` parameter. Turns out, if you're logged into a different shop, this shop never gets created but we're still passed the shop domain of the user. A little digging reveals that with user token auth enabled, it will authenticate that user, but it seems to skip cross checking this against `session[:shop_id]`.

The fix for us was to simply `return` out of `after_authenticate_job` if we couldn't find a shop with the domain passed in. The app continues on and eventually realizes the shop and the user in the cookie don't match and the merchant is redirected to the login page. However, it should never be the case that the `user_id` and `shop_id` within the session don't match the same Shopify domain.

In summary, fixing this issue would improve two things:
- `config.after_authenticate_job` could reliably pass a `shop_domain` that is guaranteed to exist
-  A merchant can install the same app while being logged into another store without having to be redirected to login

**Comment by paulomarg** [comment]
Hi @nickmealey, thank you for reporting this! We will look into improving this behaviour if we can.
